src/_posts/databases/postgresql/2000-01-01-troubleshooting.md

```diff
@@ -0,0 +1,33 @@
+---
+title: Troubleshooting Scalingo for PostgreSQL
+nav: Troubleshooting
+modified_at: 2023-12-01 00:00:00
+tags: databases postgresql addon
+index: 9
+---
+
+## Identifying Performances Issues
+
+
+## Understanding Idle Queries
+
+PostgreSQL® tends to use all the available memory if there is enough indices to
+fill the memory. If there is too many indices to fit into the memory, some of
+them are stored on the disk. In this situation, queries needing these indices
+will be slowed down. PostgreSQL® first needs to load the indices from the disk
+into the RAM which takes some time. The memory usage on the "Metrics" tab of
+your web dashboard would always be at 100% in such situation.
+
```

Comment:
## Peak of I/O

In the metrics visible in your database dashboard, you may notice I/O peaks, particularly for read operations. Those peaks can often be attributed to a few common factors. One primary reason could be insufficient memory. When the database cannot hold enough of its working set in memory, it resorts to frequent disk reads and writes, leading to high I/O. Another potential cause is executing queries that retrieve large amounts of data, especially if these operations are not optimized or if the data exceeds the available memory. Additionally, a lack of proper indexing can lead to inefficient query execution, forcing PostgreSQL to perform full table scans instead of quick index lookups, significantly increasing I/O load.

Reply:
Added in https://github.com/Scalingo/documentation/pull/2389/commits/d9c945ae6290165b02d1de051cc66238e6da8d11